Problems I faced with Postgress when tried to connect from psycopg2:

1- psycopg2.OperationalError: could not connect to server: No such file or directory
	is the server running locally and accepting
        connections on Unix domain socket "/tmp/.s.PGSQL.5432"?
        
   Solution: After further digging, found out that the socket in my installation is avialable in "/var/run/postgresql/.s.PGSQL.5432" path while psycopg2 is looking for the socket in 
   "/tmp/.s.PGSQL.5432"
   Therefore, I accessed the following file:
	"/etc/postgresql/12/main/postgres.conf" and commented 
	unix_socket_directories = '/var/run/postgresql' then added the following
	unix_socket_directories = '/tmp/'
This resolved the problem


2- when executing: >>psycopg.connect('dbname=example user=postgress') received the following error: psycopg2.OperationalError: FATAL:  Peer authentication failed for user "postgres"

Solution: 
1- execute the following command to access the authentication configuration file:
$ sudo subl /etc/postgresql/12/main/pg_hba.conf 

2- the above will open the config file in sublime. change the following two lines:
host    all             all             127.0.0.1/32            md5
host    all             all             127.0.0.1/32            md5

to be:

host    all             all             127.0.0.1/32            trust
host    all             all             127.0.0.1/32            trust

3- Restart postgres services:
$ sudo systemctl restart postgresql
To make sure it is restarted: execute the following command and see the 'active' time
$ sudo systemctl status postgresql.service


The above resolved the peer auth issue. BTW, trust means all connections will be accepted. This was a temporary solution as I want to concentrate more on postgres learning rather than auth issues!!



ls -lA /var/run/postgresql

